header/p2p/peer_tracker.go

```diff
@@ -190,3 +194,25 @@ func (p *peerTracker) stop() {
 		<-p.done
 	}
 }
+
+// blockPeer blocks a peer on the networking level and removes it from the local cache.
+func (p *peerTracker) blockPeer(pID peer.ID) {
+	// add peer to the blacklist, so we can't connect to it in the future.
+	err := p.connGater.BlockPeer(pID)
+	if err != nil {
+		log.Errorw("header/p2p: blocking peer failed", "err", err, "pID", pID)
+	}
+	// close connections to peer.
+	err = p.host.Network().ClosePeer(pID)
+	if err != nil {
+		log.Errorw("header/p2p: closing connection with peer failed", "err", err, "pID", pID)
+	}
+
+	log.Warnw("header/p2p: blocked peer", "pID", pID)
```

Comment:
would be good to log reason too but we can do in separate PR

Reply:
we can make an issue for it since it would break method, lets just keep retry to retry

Reply:
ok, sure